scispacy/hyponym_detector.py

```diff
@@ -70,7 +72,10 @@ def expand_to_noun_compound(self, token: Token, doc: Doc):
 
         end = token.i + 1
         while True:
-            previous = doc[end]
+            try:
```

Comment:
Instead of using `try/except` here, could you get the length of the doc before this while loop (`len(doc)`) and check against it each time? It's more explicit and doesn't silently catch other index errors that may be thrown e.g within spacy.

Reply:
Sorry for the delay.  Should have time to make these updates shortly.

Reply:
No problem, contributions are always welcome regardless of timeframe :)